### PR TITLE
Working html site

### PR DESCRIPTION
**Pull request by nickclifford904**

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (ashleycoles) [inline .idea/.gitignore:1]: Could you see if you can remove this entire .idea folder? It's just used by phpstorm to track your project and doesn't need to be in the repo

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/** create a function to display each chair and all their details from the database */
+
+function displayChairs($query): string
+{
+    $result = '';
+    foreach ($query as $chair) {
+        '<div class="chairtitle">';
+        $result .='<h3>' . $chair['chairname'] . '</h3></div>';
+        '<div class="chairinfo">';
```
Comment (ashleycoles) [inline functions.php]: These two divs are not actually being displayed, since they are not getting concatenated into your $result variable. This results in some malformed HTML since the closing divs are displayed, but not the opening.

Popping `$result .=` before each line should sort it!

File: index.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?php require 'functions.php'?>
+
+<h1>The Chair Collector</h1>
```
Comment (ashleycoles) [inline index.php]: Don't forget about the rest of the HTML you need in here!

```<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>title</title>
  </head>
  <body>
  
  </body>
</html>```

File: index.php [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+<?php
+
+$db = connectdb();
+$query = $db->prepare('SELECT * FROM `chaircollection`;');
```
Comment (ashleycoles) [inline index.php]: Would suggest removing the * and being more specific with which fields you want returned from the query

File: tests/tests.php [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public function testDisplayChairs_success1()
+    {
+        $test_array = [[
+            'chairname' => 'highchair',
+            'designer' => 'gary',
+            'designyear' => '2001'
+        ]];
+        $result = displayChairs($test_array);
+        $expected = '<h3>highchair' . '</h3>' . '<span>Designer: gary' . '</span><br/>' . '<span>Year: 2001' . '</span><br/>';
```
Comment (ashleycoles) [inline tests/tests.php]: Bear in mind that once you tweak the two div tags that aren't display you will need to update your expected result here to match!

File: tests/tests.php [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        $this->assertEquals($result, $expected);
+    }
```
Comment (ashleycoles) [inline tests/tests.php:19]: How about a malformed test? What happens if the displayChairs() functions is given a boolean for example?

Also a failure test could go here, what happens if displayChairs is given an empty array? IE no results from the database?

File: functions.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    $db ->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
+}
+
+/** create a function to display each chair and all their details from the database */
+
+function displayChairs($query): string
+{
+    $result = '';
+    foreach ($query as $chair) {
```
Comment (ashleycoles) [inline functions.php:22]: What happens if there are no chairs in the database? Displaying an error message here could be a good idea!

**Review by ashleycoles — CHANGES_REQUESTED** [review]
A few changes to keep you busy! Good work!
